api/src/utils/apply-diff.ts

```diff
@@ -0,0 +1,305 @@
+import { Field, Relation, SchemaOverview } from '@directus/shared/types';
+import { Knex } from 'knex';
+import { CollectionsService, FieldsService, RelationsService } from '../services';
+import { ActionEventParams, Collection, MutationOptions, Snapshot, SnapshotDiff, SnapshotField } from '../types';
+import { getSchema } from './get-schema';
+import getDatabase from '../database';
+import { applyChange, Diff, DiffDeleted, DiffNew } from 'deep-diff';
+import { cloneDeep, merge, set } from 'lodash';
+import logger from '../logger';
+import emitter from '../emitter';
+
+type CollectionDelta = {
+	collection: string;
+	diff: Diff<Collection | undefined>[];
+};
+
+export async function applyDiff(
```

Comment:
This could do with a unit test if at all possible 👀

Reply:
I believe it is still technically tested by `apply-snapshot.test.ts` since we're extracting `applySnapshot`'s inner logic into `applyDiff`, albeit indirectly now. Should we move it into `apply-diff.test.ts` instead?  🤔 though I'm not sure if `applySnapshot` still requires testing after we do so (I assume that's perfectly fine?).